bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/util/JsonPatchHelper.java

```diff
@@ -80,13 +83,19 @@ public class JsonPatchHelper extends AbstractJsonPatchHelper {
     */
    private static final AtomicBoolean NEED_OBJECT_URI_MAPPINGS = new AtomicBoolean();
 
+   private Optional<CodecProvider> getBestCodecProvider(final String modelUri, final String format) {
```

Comment:
Could this instead be a static method of the `CodecProvider` interface? Something like `CodecProvider::getBestCodecProvider(Collection<? extends CodecProvider>, String, String): Optional<CodecProvider>`.

And, in fact, since usually the next step is to get the codec, it could be nice also to have `CodecProvider::getBestCodec(Collection<? extends CodecProvider>, String, String): Optional<Codec>`. I would expect other clients than the `JsonPatchHelper` could benefit from this.